client.go

```diff
@@ -618,11 +668,26 @@ func (c *Client) connectToNode(node *Node) error {
 		}
 	}()
 
-	go func() {
+	go func() { // setClient Action function
+		// client auth
+		timer := time.NewTimer(waitForChallengeTimeout)
+
+		var saltAndSig stSaltAndSignature
+		select { // time out or get challenge signature
+		case <-timer.C:
+		case saltAndSig = <-c.chChallengeSignature:
+		}
+
 		req := make(map[string]interface{})
 		req["Action"] = "setClient"
 		req["Addr"] = c.Address()
 
+		// client auth
+		if len(saltAndSig.Signature) > 0 {
```

Comment:
If we change to pointer, this needs to be `!= nil`

Reply:
Yes, modified.